subprojects/core/src/integTest/groovy/org/gradle/api/tasks/LambdaInputsIntegrationTest.groovy

```diff
@@ -121,18 +121,19 @@ class LambdaInputsIntegrationTest extends AbstractIntegrationSpec implements Dir
         """
 
         buildFile.makeOlder()
+        def nonCacheableInputsReason = "Non-cacheable inputs: property 'action' was implemented by a Java lambda"
```

Comment:
Shall we print which lambda?

Reply:
Should we also recommend here to use an anonymous inner class?

Reply:
By which lambda you mean which type? I wouldn't do this here since we already point to the property. So it shouldn't be too hard to find it out.

We already recommend to use an anonymous inner class, the test just hasn't been asserting it. Now it does.

Reply:
> So it shouldn't be too hard to find it out.

If you got the message that `myTask.action` was a lambda, how would you go about finding out what's happening as a user?

Reply:
We are printing the implementation class now.